x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -111,6 +114,21 @@ public JwtRealm(final RealmConfig realmConfig, final SSLService sslService, fina
         this.jwtCache = this.buildJwtCache();
         this.jwtCacheHelper = (this.jwtCache == null) ? null : new CacheIteratorHelper<>(this.jwtCache);
 
+        // Validate this realm uses a principal claim that is in the allowlist for all JWT realms
+        if (this.jwtRealms.getPrincipalClaimNames().contains(this.claimParserPrincipal.getClaimName()) == false) {
+            throw new SettingsException(
+                "JWT Realm setting ["
+                    + realmConfig.getSetting(JwtRealmSettings.CLAIMS_PRINCIPAL.getClaim())
+                    + "] value ["
+                    + this.claimParserPrincipal.getClaimName()
+                    + "] is not in the allow-list JWT Realms setting ["
+                    + JwtRealmsSettings.PRINCIPAL_CLAIMS_SETTING.getRawKey()
+                    + "] value ["
+                    + String.join(",", this.jwtRealms.getPrincipalClaimNames())
+                    + "]"
+            );
+        }
```

Comment:
I don't understand why we think this is a necessary check. I don't think we want it.

It introduces a new constrains that didn't exist before this PR, even though this PR is aiming to relax constraints. I really don't understand the reasoning.

Reply:
Removed the check.